Build is failing due to name mangling

The build is failing:

```
X check build
> Run make check
python -m build
...
make check
...
Successfully built awscli_login-0.2b2.dev63.tar.gz and awscli_login-0.2b2.dev63-py3-none-any.whl
twine check --strict dist/awscli_login-0.2b2.dev63-py3-none-any.whl dist/awscli-login-0.2b2.dev63.tar.gz
ERROR    InvalidDistribution: Cannot find file (or expand pattern):             
         'dist/awscli-login-0.2b2.dev63.tar.gz'                                 
make: *** [Makefile:65: .twinecheck] Error 1
```
https://github.com/techservicesillinois/awscli-login/actions/runs/8758907346/job/24264968452?pr=190

The build is failing because the Makefile expects the source distribution filename (tarball) to have a hypen (_awscli-login_) instead of an underscore (_awscli_login_). Looking through the [release history](https://test.pypi.org/project/awscli-login/0.1.0a7.dev21/#history) for the awscli-login package on PyPI it appears the source distribution filename has always contained a hypen. This does NOT conform to [PEP 625](https://peps.python.org/pep-0625/#specification) which states the source distribution filename may contain only a single hypen separating the distribution name from the version:

> The name of an sdist should be `{distribution}-{version}.tar.gz`.
> 
> * `distribution` is the name of the distribution as defined in [PEP 345](https://peps.python.org/pep-0345/), and normalised as described in [the wheel spec](https://packaging.python.org/en/latest/specifications/binary-distribution-format/) e.g. 'pip', 'flit_core'.
> * `version` is the version of the distribution as defined in [PEP 440](https://peps.python.org/pep-0440/), e.g. 20.2, and normalised according to the rules in that PEP.
> 
> An sdist must be a gzipped tar archive in pax format, that is able to be extracted by the standard library tarfile module with the open flag 'r:gz'.
> 
> Code that produces an sdist file MUST give the file a name that matches this specification. The specification of the build_sdist hook from [PEP 517](https://peps.python.org/pep-0517/) is extended to require this naming convention.
> 
> Code that processes sdist files MAY determine the distribution name and version by simply parsing the filename, and is not required to verify that information by generating or reading the metadata from the sdist contents.
>
> Conforming sdist files can be recognised by the presence of the .tar.gz suffix and a single hyphen in the filename. Note that some legacy files may also match these criteria, but this is not expected to be an issue in practice. See the “Backwards Compatibility” section of this document for more details.

`setuptools` recently implemented [PEP 625](https://peps.python.org/pep-0625/#specification) in version [v69.3.0](https://setuptools.pypa.io/en/latest/history.html#v69-3-0) on April 12, 2024. The changes are documented [here](https://packaging.python.org/en/latest/specifications/binary-distribution-format/#escaping-and-unicode):

> As the components of the filename are separated by a dash (-, HYPHEN-MINUS), this character cannot appear within any component. This is handled as follows:
> 
> * In distribution names, any run of -_. characters (HYPHEN-MINUS, LOW LINE and FULL STOP) should be replaced with _ (LOW LINE), and uppercase characters should be replaced with corresponding lowercase ones. This is equivalent to regular [name normalization](https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization) followed by replacing - with _. Tools consuming wheels must be prepared to accept . (FULL STOP) and uppercase letters, however, as these were allowed by an earlier version of this specification.
> * Version numbers should be normalised according to the [Version specifier specification](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers). Normalised version numbers cannot contain -.
> * The remaining components may not contain - characters, so no escaping is necessary.



The code in the `Makefile` that implemented this feature was introduced in commit 9f64fc42 on Feb 5, 2024.

```Makefile
PACKAGE_NAME := awscli-login
MODULE_NAME := awscli_login
...
WHEEL := $(MODULE_NAME)-$(VERSION)-py3-none-any.whl
SDIST := $(PACKAGE_NAME)-$(VERSION).tar.gz
RELEASE := dist/$(WHEEL) dist/$(SDIST)
...
build: $(RELEASE)
$(RELEASE): docs/readme.rst pyproject.toml MANIFEST.in $(SRCS)
	python -m build
```

To resolve the issue MODULE_NAME name should be used instead of PACKAGE_NAME for the SDIST in the Makefile.

--- a/.github/workflows/cicd.yml
+++ b/.github/workflows/cicd.yml
@@ -153,8 +153,16 @@ jobs:
         python -m pip install --upgrade --upgrade-strategy eager tox wheel
 
     - name: Update Python wheel cache
-      if: steps.cache.outputs.valid != 'true'
+      if: steps.cache.outputs.valid != 'true' && matrix.os != 'macos-latest'
       run: make cache
+
+    # Compile lxml on ARM macOS systems to avoid bad wheels on PyPI.
+    # See issue #189.
+    - name: Update Python wheel cache (macOS)
+      if: steps.cache.outputs.valid != 'true' && matrix.os == 'macos-latest'
+      run: make cache
+      env:
+        PIP_NO_BINARY: 'lxml'
 
     - name: Install
       if: steps.tox.outputs.valid != 'true'

--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ HTML := htmlcov/index.html
 TOX := tox -e wheel -qq --skip-pkg-install
 TOX_ENV := .tox/wheel/pyvenv.cfg
 WHEEL := $(MODULE_NAME)-$(VERSION)-py3-none-any.whl
-SDIST := $(PACKAGE_NAME)-$(VERSION).tar.gz
+SDIST := $(MODULE_NAME)-$(VERSION).tar.gz
 RELEASE := dist/$(WHEEL) dist/$(SDIST)
 PIP := python -m pip install --upgrade --upgrade-strategy eager
 

--- a/docs/readme/body.rst
+++ b/docs/readme/body.rst
@@ -405,3 +405,16 @@ the location of a Windows install of awscli::
 To remedy this issue please ensure that the location where the
 awscli is installed in the WSL comes before the location of the
 Windows install in the WSL PATH environment variable.
+
+
+lxml import errors on macOS
+---------------------------
+
+On M1 and M2 Apple MacBooks you may receive the following error at runtime::
+
+    ImportError: dlopen(/Users/ddriddle/.pyenv/versions/3.8.16/lib/python3.8/site-packages/lxml/etree.cpython-38-darwin.so, 0x0002): symbol not found in flat namespace '_exsltDateXpathCtxtRegister'
+
+This can be fixed by removing and compiling lxml::
+
+    pip uninstall lxml
+    PIP_NO_BINARY=lxml pip install lxml

--- a/src/integration_tests/tests/build.bats
+++ b/src/integration_tests/tests/build.bats
@@ -11,6 +11,6 @@ setup() {
 }
 
 @test "Install from source distribution" {
-    run pip install --no-deps ../../dist/awscli-login*.tar.gz
+    run pip install --no-deps ../../dist/awscli_login*.tar.gz
     assert_success
 }

